src/openslide-vendor-philips-isyntax.c

```diff
@@ -0,0 +1,397 @@
+    // tile size
+    int64_t tw = libisyntax_get_tile_width(isyntax);
+    int64_t th = libisyntax_get_tile_height(isyntax);
+
+    // Openslide cache
+    g_autoptr(_openslide_cache_entry) cache_entry = NULL;
+    uint32_t *tiledata = _openslide_cache_get(osr->cache,
+                                              pi_level, tile_col, tile_row,
+                                              &cache_entry);
+    if (!tiledata) {
+        int scale = libisyntax_level_get_scale(pi_level->isyntax_level);
+        ASSERT_OK(libisyntax_tile_read(isyntax, data->cache->cache, scale, tile_col, tile_row, &tiledata));
+        annotate_tile(tiledata, scale, tile_col, tile_row, tw, th);
+
+        _openslide_cache_put(osr->cache, pi_level, tile_col, tile_row,
+                             tiledata, tw * th * 4,
+                             &cache_entry);
+    }
+
+    // draw it
+    g_autoptr(cairo_surface_t) surface =
+            cairo_image_surface_create_for_data((unsigned char *) tiledata,
+                                                CAIRO_FORMAT_ARGB32,
+                                                tw, th, tw * 4);
+    cairo_set_source_surface(cr, surface, 0, 0);
+    // https://lists.cairographics.org/archives/cairo/2012-June/023206.html
+    // Those are the operators that are observed to work:
+    //    w CAIRO_OPERATOR_SATURATE (current_cairo_operator, aka default OpenSlide)
+    //    w CAIRO_OPERATOR_OVER ("This operator is cairo's default operator."),
+    //    w CAIRO_OPERATOR_DEST_OVER,
+    // SATURATE takes ~12sec to read a dummy slide (forcing all tiles to not exist), OVER & DEST_OVER take 3.5 sec
+    // for same setup. Selecting OVER as the Cairo's default operator, the three outputs are identical.
+    cairo_operator_t current_cairo_operator = cairo_get_operator(cr);
+    cairo_set_operator(cr, CAIRO_OPERATOR_OVER);
+    cairo_paint(cr);
+    cairo_set_operator(cr, current_cairo_operator);
+    return true;
+}
+
+static void add_float_property(openslide_t *osr, const char* property_name, float value) {
+    g_hash_table_insert(osr->properties, g_strdup(property_name),
+                        _openslide_format_double(value));
+}
+
+static bool philips_isyntax_open(
+        openslide_t *osr,
+        const char *filename,
+        struct _openslide_tifflike *tl G_GNUC_UNUSED,
+        struct _openslide_hash *quickhash1 G_GNUC_UNUSED,
+        GError **err) {
+    // Do not allow multithreading in opening.
+    // https://docs.gtk.org/glib/method.Mutex.init.html:
+    //   "It is not necessary to initialize a mutex that has been statically allocated."
+    static GMutex static_open_mutex;
+    g_mutex_lock(&static_open_mutex);
+    static bool threadmemory_initialized = false;
+    if (!threadmemory_initialized) {
+        // TODO(avirodov): review libisyntax_init() thread safety, if thread safe, remove the mutex here.
+        libisyntax_init();
```

Comment:
FYI https://github.com/amspath/libisyntax/pull/22 "Global mutex with atomics". I will resolve this when the PR is accepted and I update `openslide-vendor-philips-isyntax.c` to not have mutex here.

Reply:
Done, though the mutex is still used for the shared cache. This will get removed when I change the cache api, but the libisyntax_init() issue is resolved.